Flit package

mikebentley15 opened the pull request:
Add the FLiT package along with py-toml

junghans commented on var/spack/repos/builtin/packages/flit/package.py:
```diff
@@ -0,0 +1,67 @@
+    version('2.0-alpha.1', '62cf7784bcdc15b962c813b11e478159')
+    # FIXME: fix install and build to handle the old version, which is not
+    #        installable
+    # version('1.0.0',       '27763c89b044c5e3cfe62dd319a36a2b')
```
Just use:
```
conflicts("@:2", msg="Only can build version 2.0 and up")
```

mikebentley15 replied:
Okay, thanks.  Done

mikebentley15 replied:
I added this and then was no longer able to install flit, saying that 2.0-alpha.1 conflicts with @:2.  Is this the desired behavior of spack?  For now, I changed it to be conflicts("@:1.999", ...) so that I could verify that it works and get it installed on my laptop.

mikebentley15 commented:
I addressed your review comment.  I can't change the status of your request or anything of the sort.  It doesn't seem I have enough permissions.  Can you please merge in this pull request?

alalazo commented on var/spack/repos/builtin/packages/flit/package.py:
```diff
@@ -0,0 +1,68 @@
+            make(*args)
+        else:
+            # FIXME: build the old FLiT
+            raise NotImplementedError('Only can build version 2.0 and up')
```
The conditional here is redundant, can you remove it?

mikebentley15 replied:
done

alalazo commented on var/spack/repos/builtin/packages/flit/package.py:
```diff
@@ -0,0 +1,68 @@
+            args = self.install_targets[:]
+            args.append('PREFIX=%s' % prefix)
+            make(*args, parallel=False)
+        else:
```
Same redundancy here.

mikebentley15 replied:
done

alalazo commented on var/spack/repos/builtin/packages/flit/package.py:
```diff
@@ -0,0 +1,68 @@
+            raise NotImplementedError('Only can build version 2.0 and up')
+
+    def install(self, spec, prefix):
+        'Install instructions based on the version'
```
This is not a strict requirement, but everywhere else we use [triple quoted strings](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str) for docstrings. Could you please change this?

mikebentley15 replied:
done

alalazo commented on var/spack/repos/builtin/packages/flit/package.py:
```diff
@@ -0,0 +1,68 @@
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+'Spack package specification for the FLiT tool'
```
No need for this string.

mikebentley15 replied:
done

alalazo commented on var/spack/repos/builtin/packages/py-toml/package.py:
```diff
@@ -0,0 +1,56 @@
+# See the Spack documentation for more information on packaging.
+# If you submit this package back to Spack as a pull request,
+# please first remove this boilerplate and all FIXME comments.
+#
```
> \# If you submit this package back to Spack as a pull request,
￼  # please first remove this boilerplate and all FIXME comments.

:roll_eyes:

mikebentley15 replied:
Sorry, I forgot to go back to the py-toml package and clean it up.  I was first wanting to get it to work.  It is now fixed.

alalazo commented on var/spack/repos/builtin/packages/py-toml/package.py:
```diff
@@ -0,0 +1,56 @@
+
+
+class PyToml(PythonPackage):
+    """FIXME: Put a proper description of your package here."""
```
Also here.

mikebentley15 replied:
done

alalazo commented on var/spack/repos/builtin/packages/py-toml/package.py:
```diff
@@ -0,0 +1,56 @@
+class PyToml(PythonPackage):
+    """FIXME: Put a proper description of your package here."""
+
+    # FIXME: Add a proper url for your package's homepage here.
```
And here.

mikebentley15 replied:
done

alalazo commented on var/spack/repos/builtin/packages/flit/package.py:
```diff
@@ -0,0 +1,59 @@
+    def build(self, spec, prefix):
+        """Build instructions based on the version"""
+        args = self.build_targets[:]
+        make(*args)
```
Probably you can completely remove `build` (the current implementation looks the same as the default one).

alalazo commented on var/spack/repos/builtin/packages/flit/package.py:
```diff
@@ -0,0 +1,59 @@
+        """Install instructions based on the version"""
+        args = self.install_targets[:]
+        args.append('PREFIX=%s' % prefix)
+        make(*args, parallel=False)
```
It's fine with me to maintain this function. Just for your information, almost the same behavior could have been implemented as:
```python
@property
def install_targets(self):
    return ['install', 'PREFIX=%s' % prefix]
```
and eventually:
```python
parallel =False
```
at class level (the only difference being that also the build phase would run serially).

mikebentley15 commented:
@junghans do you approve of my change now?  I addressed your review comment.
